src/infra/sslQueryRepo.go

```diff
@@ -0,0 +1,194 @@
+package infra
+
+import (
+	"errors"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/speedianet/sam/src/domain/entity"
+	"github.com/speedianet/sam/src/domain/valueObject"
+	infraHelper "github.com/speedianet/sam/src/infra/helper"
+)
+
+const olsHttpdConfigPath = "/usr/local/lsws/conf/httpd_config.conf"
+
+type SslQueryRepo struct{}
+
+func (repo SslQueryRepo) splitSslCertificate(
+	sslCertContent string,
+) ([]entity.SslCertificate, error) {
+	certificates := []entity.SslCertificate{}
+
+	sslCertContentSlice := strings.SplitAfter(sslCertContent, "-----END CERTIFICATE-----\n")
+	for _, sslCertContentStr := range sslCertContentSlice {
+		certificate, err := entity.NewSslCertificate(sslCertContentStr)
+		if err != nil {
+			return certificates, err
+		}
+		certificates = append(certificates, certificate)
+	}
+
+	return certificates, nil
+}
+
+func (repo SslQueryRepo) GetHttpdVhostsConfig() (map[string]string, error) {
```

Comment:
Cara, eu acho a existência da sua função GetHttpdVhostsConfig() um problema. Primeiro, ela é uma função difícil de ler e que retorna 2 valores num map[string]string que tu só consegue entender se ler múltiplas vezes pra sacar que a primeira chave é o vhost e a segunda o file path da config.

Segundo, ela é usada duas vezes: uma vez na GetVhostConfigFilePath() que nada mais é que uma função que filtra a GetHttpdVhostConfig() para pegar apenas o config file path do vhost solicitado e uma vez pela GetSslPair para fazer um loop nos config files.

Nessa aplicação, o usuário vai ter uns 3 vhosts em média no máximo, o domínio principal e 1 ou 2 sub-domínios, pois o projeto é para conteinerização de uma única aplicação ou micro serviço. 

Logo, a GetHttpdVhostsConfig() no meu entendimento existe para evitar abrir o arquivo httpd.conf 3x, certo? É uma busca por uma eficiência que não resulta em ganho de velocidade ou consumo perceptível e que custa muito em legibilidade.

Você tem a função GetVhostConfigFilePath(). Leva a lógica de ler o httpd.conf pra ela em busca do config path do vhost e crie uma função chamada GetVhost(). Assim lá na sua GetSslPair, primeiro você pega a lista de vhost e depois lê cada um dos vhosts em busca dos SSLs.

Vai obrigar a leitura do arquivo httpd.conf 3x ao invés de 1x, vai, mas seu código vai ficar mais óbvio e claro. Não troque legibilidade por economia de I/O ou ciclo de CPU a não ser que o benefício reflita no tempo de resposta do app e ultrapasse o SLA.

Primeiro você constrói o produto, depois você o torna mais eficiente. Tentar fazer os dois ao mesmo tempo, tu vai estar resolvendo um problema que não existe. As chances são de que o produto vai mudar antes de você precisar tornar algo nele mais eficiente e quando for tornar algo mais eficiente, você já terá muito feedback do que realmente precisa ser eficiente e QUANTO mais eficiente ele precisa ser.

Reply:
Estou com dificuldade de visualizar isso, então para evitar a implementação errada, vou comentar sobre. Me corrija ao longo do comentário, por gentileza.

Meu `GetSslPairs` tem a função de trazer todos os certificados SSLs relativo ao `httpd.conf` do usuário. Mas, esse método é um retorno para o _use case_ que faz a listagem desses certificados, ou seja, o usuário não me informa qual o `vhost` que ele deseja filtrar. Dito isso, necessito abrir o `httpd.conf` para saber quais são os `vhosts` disponíveis. O método `GetHttpdVhostsConfig()` tem essa exata função. Ele não foi criado para ser eficiente e/ou performático, ele foi criado para saciar essa necessidade que meu repositório tem de ter acesso a todos os `vhosts` disponíveis no `httpd.conf`.

Não tenho problema nenhum em abrir o arquivo várias vezes, apenas não há como abrir ele 3x (caso o usuário tenha 3 `vhosts`) para buscar cada `vhost`, uma vez que eu não tenho acesso a essa lista de `vhosts`, eu só tenho acesso quando eu efetivamente abro o `httpd.conf`. Ou seja, se eu for criar um método que abra o `httpd.conf` e lista todos os meus `vhosts`, eu já vou ter aberto meu `httpd.conf` uma vez e não três.

Quanto ao método `GetVhostConfigFilePath()` ele filtra o `vhost` que quero simplesmente para não ficar repetindo as mesmas linhas de código que estão dentro do método. Especialmente porque eu utilizo ele no meu `SslCmdRepo`, ou seja, fora do arquivo onde essa função está construída.

O fluxo que foi construído em volta desses métodos é o seguinte:
- Usuário envia requisição `GET - /v1/ssl` sem parâmetros;
- Meu _controller_ chama o _use case_ `GetSslPairs` que tem a única função de chamar a função `GetSslPairs` do meu repositório _query_;
    - Não tenho nenhum parâmetro de filtro ou de busca, portanto preciso listar e não filtrar.
- Abro o arquivo `httpd.conf` e listo todos os `vhosts` disponíveis e onde está seu arquivo de configuração;
    - **Observação:** preciso saber o caminho do arquivo, se não num consigo abrir o arquivo de configuração do `vhost`.
- Crio um laço nessa lista e trago o conteúdo de cada um dos arquivos `vhost` para montar minha lista de `SslPair` e retornar para meu _controller_.
- Retorno a lista para o usuário.

O `GetHttpdVhostsConfig()` tem essa única funcionalidade: abrir o `httpd.conf` e listar os `vhosts` para mim. O `GetVhostConfigFilePath()` tem a única função de filtrar essa lista e retornar o `vhost` desejado apenas quando necessário. Ambas as funções tem objetivos diferentes.